tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Editing/CodeownersEditor.cs

```diff
@@ -0,0 +1,106 @@
+            {
+                updatedEntry = CodeownersHelper.UpdateCodeownersEntry(matchingEntry, serviceOwners, sourceOwners, true);
+                codeownersContent = CodeownersHelper.AddCodeownersEntryToFile(codeownersEntries, codeownersContent, updatedEntry, true);
+            }
+            else
+            {
+                updatedEntry = CodeownersHelper.CreateCodeownersEntry(normalizedPath, serviceLabel, serviceOwners, sourceOwners, isMgmtPlane);
+                codeownersContent = CodeownersHelper.AddCodeownersEntryToFile(codeownersEntries, codeownersContent, updatedEntry, false);
+            }
+            return updatedEntry;
+        }
+
+        /// <summary>
+        /// Removes owners from an existing codeowners entry, based on path or service label.
+        /// Updates the related codeowners content string to include the modified codeowners entry.
+        /// Handles all normalization internally.
+        /// </summary>
+        /// <param name="path">The path for the codeowners entry (optional if serviceLabel is provided).</param>
+        /// <param name="serviceLabel">The service label for the codeowners entry (optional if path is provided).</param>
+        /// <param name="serviceOwnersToRemove">List of service owners to remove.</param>
+        /// <param name="sourceOwnersToRemove">List of source owners to remove.</param>
+        /// <param name="isMgmtPlane">Whether this is a management-plane entry.</param>
+        /// <returns>The updated CodeownersEntry after removal.</returns>
+        public CodeownersEntry RemoveOwners(string path = "", string serviceLabel = "", List<string> serviceOwnersToRemove = null, List<string> sourceOwnersToRemove = null, bool isMgmtPlane = false)
+        {
+            string normalizedPath = path;
+            if (!string.IsNullOrWhiteSpace(path))
+            {
+                normalizedPath = CodeownersHelper.NormalizePath(path);
+            }
+
+            var codeownersContentList = codeownersContent.Split('\n').ToList();
+
+            var (startLine, endLine) = isMgmtPlane
+                ? CodeownersHelper.FindBlock(codeownersContent, standardManagementCategory)
+                : CodeownersHelper.FindBlock(codeownersContent, standardServiceCategory);
+
+            var codeownersEntries = CodeownersParser.ParseCodeownersEntries(codeownersContentList, AzureWriteTeamsBlobUrl, startLine, endLine);
+            var matchingEntry = CodeownersHelper.FindMatchingEntry(codeownersEntries, normalizedPath, serviceLabel);
+
+            if (matchingEntry == null)
+            {
+                throw new InvalidOperationException("No matching entry found to remove owners from.");
+            }
+            var updatedEntry = CodeownersHelper.UpdateCodeownersEntry(matchingEntry, serviceOwnersToRemove, sourceOwnersToRemove, false);
+            codeownersContent = CodeownersHelper.AddCodeownersEntryToFile(codeownersEntries, codeownersContent, updatedEntry, true);
```

Comment:
In my opinion the editor should hide the file details instead of updating it on the side. Then you can do `editor.AddEntry()` and `editor.Write()` making it easier to test the former and more efficient to run if multiple codeowners entries need to be added in one pass.